File: src/main/java/net/igneo/icv/entity/stonePillar/StonePillarModel.java
package net.igneo.icv.entity.stonePillar;

import net.igneo.icv.ICV;
import net.minecraft.resources.ResourceLocation;
import software.bernie.geckolib.model.GeoModel;

public class StonePillarModel extends GeoModel<StonePillarEntity> {
    private final ResourceLocation model = new ResourceLocation(ICV.MOD_ID, "geo/placeholder.geo.json");
    private final ResourceLocation texture = new ResourceLocation(ICV.MOD_ID, "textures/entity/placeholder.png");
    private final ResourceLocation animations = new ResourceLocation(ICV.MOD_ID, "animations/placeholder.animation.json");
    @Override
    public ResourceLocation getModelResource(StonePillarEntity surfWaveEntity) {
        return model;
    }

    @Override
    public ResourceLocation getTextureResource(StonePillarEntity surfWaveEntity) {
        return texture;
    }

    @Override
    public ResourceLocation getAnimationResource(StonePillarEntity surfWaveEntity) {
        return animations;
    }
}

File: src/main/java/net/igneo/icv/entity/boots/stonePillar/StonePillarModel.kt
package net.igneo.icv.entity.boots.stonePillar

import net.igneo.icv.ICV
import net.minecraft.resources.ResourceLocation
import software.bernie.geckolib.model.GeoModel

class StonePillarModel : GeoModel<StonePillarEntity?>() {
    private val model = ResourceLocation(ICV.MOD_ID, "geo/placeholder.geo.json")
    private val texture = ResourceLocation(ICV.MOD_ID, "textures/entity/placeholder.png")
    private val animations = ResourceLocation(ICV.MOD_ID, "animations/placeholder.animation.json")

    override fun getModelResource(surfWaveEntity: StonePillarEntity?): ResourceLocation {
        return model
    }

    override fun getTextureResource(surfWaveEntity: StonePillarEntity?): ResourceLocation {
        return texture
    }

    override fun getAnimationResource(surfWaveEntity: StonePillarEntity?): ResourceLocation {
        return animations
    }
}
